### PR TITLE
chore: Remove deprecated MFA streaming methods

### PR DESCRIPTION
**Pull request by codingllama**
Remove MFA streaming methods deprecated by #32516.

#20343

### CODE REVIEW

File: lib/auth/auth.go [hunk 1]
```diff
@@ -3428,10 +3428,6 @@ type newMFADeviceFields struct {
 	// tokenID is the ID of a reset/invite/recovery/privilege token.
 	// It is generally used to recover the TOTP secret stored in the token.
 	tokenID string
-	// totpSecret is a secret shared by client and server to generate totp codes.
-	// Field can be empty to get secret by "tokenID".
-	// DELETE IN 16. Only used by the streaming AddMFADevice RPC. (codingllama)
-	totpSecret string
```
Comment (codingllama) [inline lib/auth/auth.go:3434]: I had another look at this one and the comment seems correct - the field is read by `registerTOTPDevice` but never written.

**Comment by espadolini** [comment]
Why do we need the empty messages?

**Comment by codingllama** [comment]
> Why do we need the empty messages?

I'm keeping the RPCs with a custom error message, so we need to retain the request/responses.

**Comment by codingllama** [comment]
I forgot to cleanup the clients, so I've added a couple more commits as a consequence: 280515f and 991f107. PTAL.

File: lib/auth/auth.go [hunk 2]
```diff
@@ -3115,23 +3115,19 @@ type newRegisterChallengeRequest struct {
...
+		if req.token == nil {
+			return nil, trace.BadParameter("all TOTP registrations require a privilege token")
+		}
+
 		otpKey, otpOpts, err := a.newTOTPKey(req.username)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
 
-		// TODO(codingllama): Once AddMFADeviceSync is no more all requests should
```
Comment (codingllama) [inline lib/auth/auth.go:3123]: I believe this comment meant "AddMFADevice" (the removed stream) and not "AddMFADeviceSync". The stream had a long-running memory, whereas the unary RPCs need to keep tabs of it in storage.

**Review by zmb3 — APPROVED** [review]

**Review by espadolini — APPROVED** [review]

**Pull request merged**